dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannelFactory.java

```diff
@@ -30,6 +32,8 @@ public interface DialogueChannelFactory {
     interface ChannelArgs {
         String uri();
 
+        Optional<InetAddress> resolvedAddress();
```

Comment:
is it possible for there to be multiple resolved addresses here?

Reply:
Not the way this is constructed, as it specifies a particular target here: https://github.com/palantir/dialogue/pull/2134/files#diff-1a7529217d0eef0aaa07e3e259b4907f671eab95a67619174a7a6d4707cd0bb2R49

If an empty optional is provided, this falls back to the current behavior (which is why it's safe to merge).

I could see a model where we provide both ipv4 and ipv6 for a given node on a single client, however that opens a few new cans of worms that I'd prefer we avoid. We can always add a separate field to support that sort of thing in the future it it turns out to be important.